### PR TITLE
Logging updates

### PR DESCRIPTION
**Pull request by grybmadsci**
I'd like to propose a few changes to the logging mechanisms we use.  A summary of the changes here:

- Change flag names to be a little more precise, and to use hyphens instead of underscore (it's more standard, even if it makes the interface to gflags a little uglier for us).
- Provide better isolation between test logs and framework logs by outputting test logs to stdout and framework logs to stderr, and don't propagate test logs to framework logs.
- Other changes are cleanup/docs updates.

I'm fielding this for discussion, happy to reconsider if there are objections.  Notably, I want to make sure to get an LGTM from wallaceb before merging because this may break some of his things if he's using these flags.  Let me know what you guys think.

### CODE REVIEW

**Comment by wallacbe** [comment]
This shouldn't break me, I'm currently not explicitly setting flags.

To summarize my usage, In my phases I'm doing the following to get test output to stdout:

  logging.basicConfig()
  test.logger.info(' foo')

And I'm currently not seeing logging go to stdout in plugs

**Comment by grybmadsci** [comment]
Ah, you shouldn't have to call .basicConfig(), in fact that's probably overriding some of the things we set up internally (and might be why it's working for you)... but it won't hurt anything to do so.

File: openhtf/util/logs.py [hunk 1]
```diff
@@ -37,14 +65,15 @@
...
-                   'Logging verbosity level.')
+                   'Console log verbosity level (stderr).')
+gflags.DEFINE_boolean('quiet', False, "If True, don't output logs to stderr.")
+gflags.DEFINE_string('log-file', '', 'Filename to output logs to, if any.')
+gflags.DEFINE_enum('log-file-level', 'warning', ['debug', 'info', 'warning',
+                                                 'error', 'critical'],
+                   'Logging verbosity level for log file output.')
 
 LOGGER_PREFIX = 'openhtf'
+RECORD_LOGGER = '.'.join(LOGGER_PREFIX, 'test_record')
```
Comment (fahhem) [inline openhtf/util/logs.py]: join takes a list (broke the tests)

**Comment by fahhem** [comment]
LGTM after the tests are fixed

**Comment by fahhem** [comment]
might want to pull back on calling yourself an idiot in the commits? Also, the tests are still broken due to the output_test detecting small changes in the framework

**Pull request merged**